crates/index/src/setting.rs

```diff
@@ -0,0 +1,60 @@
+use super::Index;
+use crate::optimizing::indexing::OptimizerIndexing;
+use crate::Op;
+use base::index::IndexFlexibleOptions;
+use crossbeam::channel::{bounded, Receiver, RecvTimeoutError, Sender};
+use std::convert::Infallible;
+use std::sync::Arc;
+use std::thread::JoinHandle;
+use std::time::Duration;
+
+pub struct FlexibleOptionSyncing<O: Op> {
+    index: Arc<Index<O>>,
+}
+
+impl<O: Op> FlexibleOptionSyncing<O> {
+    pub fn new(index: Arc<Index<O>>) -> Self {
+        Self { index }
+    }
+    pub fn spawn(self) -> (Sender<Infallible>, JoinHandle<()>) {
+        let (tx, rx) = bounded(1);
+        (
+            tx,
+            std::thread::spawn(move || {
+                self.main(rx);
+            }),
+        )
+    }
+    pub fn main(self, shutdown: Receiver<Infallible>) {
+        let index = &self.index;
+        let mut old_options = index.flexible.clone();
+        let dur = Duration::from_secs(5);
+        loop {
+            let new_options = index.flexible.clone();
+            if new_options != old_options {
+                std::fs::write(
+                    index.path.join("flexible"),
+                    serde_json::to_string::<IndexFlexibleOptions>(&new_options).unwrap(),
+                )
+                .unwrap();
```

Comment:
If it's the first write to the file, `sync_all` should be called with the parent directory. We'd better not write new files under the index directory. Also, the write is not atomic so if it fails, the bgworker will panic because it reads incorrect data.